src/vs/workbench/browser/quickaccess.ts

```diff
@@ -51,14 +52,28 @@ export function getQuickNavigateHandler(id: string, next?: boolean): ICommandHan
 		quickInputService.navigate(!!next, quickNavigate);
 	};
 }
-export class EditorViewState {
+export class EditorViewStateManager extends Disposable {
```

Comment:
Wouldn't it be much easier to change this manager to have a method to open an editor transient and then track these editors for closing later? Here is how I think it could work:
- a quick picker that wants this feature already has a `EditorViewStateManager` as class member
- the picker can use the new method to open an editor transient
- the view state manager can remember which editors have been opened transient and then close them on restore

I think with that approach we are 100% sure that we only close editors that were opened from a picker. We still need to test for the editor being transient at the point we want to close it in case the user has interacted with it.

Reply:
I think that the implementation that we currently have also supports the case that the caller only wants to retain editor state but did not necessarily work with transient editors. But I'm wondering if there is any case where you want behavior like that.

Reply:
I think we should adopt `transient` also for the `file.ts@symbol` case to have a consistent solution, what do you think?

Reply:
Yes, I think so also. So do you think we should have a solution for "editor state restore" behavior without having behavior for clearing related transient editors when the restore occurs? I guess the behavior is the same if the caller ends up never opening up transient editors through the `EditorViewStateManager`.

Reply:
I think it could all be within the `EditorViewStateManager`, maybe it should be renamed to better reflect what it does:
* it has a method to open editors transient
* it has a method to clear that state
* it is being used in all pickers that open editors in the background ad-hoc

Reply:
Hmm, I'm not great with naming. How about `PickerEditorState`?

Reply:
I guess that doesn't really express that it opens transient editors.

Reply:
I was also thinking of putting it in its own file since it's currently taking up most of `quickaccess.ts` which seems strange. Not too sure where we should put the new file also.

Reply:
Okay, I'm looking into the possibility of opening a transient editor through the `EditorViewStateManager` and it might be a lot if we try to list all of the possibilities of calling `openEditor` 
https://github.com/microsoft/vscode/blob/500401c8ac5afd54326a5addcdf092db3eedc311/src/vs/workbench/services/editor/common/editorService.ts#L247-L264

Do you have a suggestion on how best to handle this? If we directly copy over even one of these editor overloads, I feel that there might be too much debt of making sure that it doesn't change. Alternatively, the caller can just notify the `EditorViewStateManager` that it opened an editor after it opens the editor itself.

Reply:
This is only for the one case of opening a transient editor in the background, no? Then I would just have 1 method `openEditorTransient` without these overloads just asking for the resource.

Reply:
I'm assuming that the user will want other options too, and also might want to specify which group they want to open it in. For example, this call is transient, but it also has other args that pertain to `ITextEditorOptions`.

https://github.com/microsoft/vscode/blob/bf43416e3b29020bec7dac51e9ce32da6a089c11/src/vs/workbench/contrib/search/browser/quickTextSearch/textSearchQuickAccess.ts#L126

Reply:
Yeah good point, I think if we wanna go down this path we need to support every call we do today in quick open that will then use the new `transient` option.

Reply:
So then does that mean that it needs an overload to support all of these?
https://github.com/microsoft/vscode/blob/500401c8ac5afd54326a5addcdf092db3eedc311/src/vs/workbench/services/editor/common/editorService.ts#L247-L250

Not too sure whether anyone would want to open a diff editor (for now), so we could always just make the args like:
```typescript
openTransientEditor(editor: IResourceEditorInput | ITextResourceEditorInput | IUntitledTextResourceEditorInput | IUntypedEditorInput, group?: IEditorGroup | GroupIdentifier | SIDE_GROUP_TYPE | ACTIVE_GROUP_TYPE | AUX_WINDOW_GROUP_TYPE): Promise<IEditorPane | undefined>;
```

Would that work?

Reply:
Yeah lets try that!